CesiumIonClient/include/CesiumIonClient/ApplicationData.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace CesiumIonClient {
+
+/**
+ *  @brief An enumeration representing the values of {@link ApplicationData::applicationMode}.
+ */
+enum ApplicationMode { CesiumIon, Saml, SingleUser };
```

Comment:
General comment about "Application": you're describing characteristics of the ion server, and that really doesn't have anything to do with an "application." I think you could come up with a more descriptive name i.e. "IonServerState".